ompi/mca/coll/base/coll_base_alltoall.c

```diff
@@ -297,42 +372,48 @@ int ompi_coll_base_alltoall_intra_bruck(const void *sbuf, int scount,
     }
 
     /* perform communication step */
-    for (distance = 1; distance < size; distance<<=1) {
-
-        sendto = (rank + distance) % size;
-        recvfrom = (rank - distance + size) % size;
+    for (distance = 1; distance < size; distance *= radix) {
+        for (i = 1; i < radix; i++) {
```

Comment:
Can we ensure radix  > 1?

Reply:
Should we check this at the start of this function and goto err_hndl if radix < 2?

Reply:
Throwing an error is probably the least surprising thing to do.